Design/Rule0018NoEventsInInternalCodeunits.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Dynamics.Nav.Analyzers.Common.AppSourceCopConfiguration;
+using Microsoft.Dynamics.Nav.CodeAnalysis;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Diagnostics;
+using Microsoft.Dynamics.Nav.CodeAnalysis.Symbols;
+using System;
+using System.Collections.Immutable;
+
+namespace BusinessCentral.LinterCop.Design {
+    [DiagnosticAnalyzer]
+    class Rule0018NoEventsInInternalCodeunits : DiagnosticAnalyzer {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create<DiagnosticDescriptor>(DiagnosticDescriptors.Rule0018NoEventsInInternalCodeunitsAnalyzerDescriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSymbolAction(new Action<SymbolAnalysisContext>(CheckInternalCodeunitForEvents), SymbolKind.Codeunit);
+        }
+
+        private void CheckInternalCodeunitForEvents(SymbolAnalysisContext symbolAnalysisContext)
+        {
+            ICodeunitTypeSymbol codeunitSymbol = symbolAnalysisContext.Symbol as ICodeunitTypeSymbol;
+            if (codeunitSymbol == null)
+            {
+                return;
+            }
+            if (codeunitSymbol.IsObsoleteRemoved || codeunitSymbol.IsObsoletePending)
+            {
+                return;
+            }
+            if (codeunitSymbol.DeclaredAccessibility != Accessibility.Internal)
+            {
+                return;
+            }
+
+            ImmutableArray<ISymbol>.Enumerator enumerator = codeunitSymbol.GetMembers().GetEnumerator();
+            while (enumerator.MoveNext())
+            {
+                ISymbol current = enumerator.Current;
+                SymbolKind kind = current.Kind;
+
+                if (kind == SymbolKind.Method)
+                {
+                    IMethodSymbol methodSymbol = (IMethodSymbol)current;
+                    if (methodSymbol.IsEvent)
```

Comment:
@rvanbekkum what about `InternalEvent`, I guess that they will get this warning as well?

Are there any more property that can be checked here?

Reply:
![image](https://user-images.githubusercontent.com/7383241/165343678-db7e3a34-f3d1-47b1-a790-951d35e0beb9.png)

So, I assume you would like to skip that one?

Reply:
Exactly, it is after all already declared as an Internal Only event... Don't you agree?

Reply:
Yes, I agree.

Reply:
I have updated the code to not raise a warning for `InternalEvent` events.

![image](https://user-images.githubusercontent.com/7383241/165384946-5bff718e-ef5c-41b6-a5d8-dacb6c102f3f.png)

Reply:
Cool Rob, that looks better! 👍